### PR TITLE
Rework e2e tests to ensure correct operation of integration tests

### LINKED ISSUE #1358: Build test bananas
There are quite a few .bna files checked in for test purposes

## Context
Binary files in git are not ideal for dealing with updates/merge conflicts

## Expected Behavior
.bna files should be created in a pretest script

## Actual Behavior
.bna files are created manually and checked in

### LINKED ISSUE #3527: Playground e2e tests are not running in Travis PRs
It would appear that the e2e tests in travis are failing out ... and not failing the builds. This needs to be investigated as a matter of urgency. The e2e tests should run, and if anything causes them to fail at any point, then the build should fail.

Evidence points to the introduction of Fabric based tests causing an unspotted regression. Proposal is to:
 - Revert to running only the web based Playground tests
 - Ensure that if any processes required for setting up the e2e tests fails, then the build fails
 - Investigate and solve the running of fabric based tests with the Playground Protractor suite
 - Issue penguins

### PR DESCRIPTION
**Pull request by nklincoln**
Signed-off-by: Nick Lincoln <nkl199@yahoo.co.uk>

closes #3527 
closes #1358

Substantial re-work of e2e tests:
 - moving parts broken out into folders to match
 - move to using ts files through all test specs and utilities
 - removal of binary files (now generating bna files in tmp dir from resources)
 - use of constants file to specify constant/global test items in all tests
 - ensuring that if publish fails, the tests (and consequently the build) fails
 - publishing of composer-playground to verdaccio to enable testing with npm playground with fabric
 - use of pm2 to run verdaccio (start/delete)
 - change of verdaccio port to prevent possible clash with other integration tests and increasing max_body_size to enable composer-playground to be published
 - breaking up of script files to make them more maintainable/understandable
 - removal of unused imports in test specs 
 - change of npm composer-playground protractor test port to be different from the local cli.js test

### CODE REVIEW

File: packages/composer-playground/e2e/data/sample-networks/basic-sample-network/lib/sample.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Sample transaction processor function.
+ * @param {org.acme.sample.SampleTransaction} tx The sample transaction instance.
+ * @transaction
+ */
+function sampleTransaction(tx) {
```
Comment (sstone1) [inline packages/composer-playground/e2e/data/sample-networks/basic-sample-network/lib/sample.js]: y u no pull in ES7/ES8 version of sample with async/await goodness?! :trollface:

File: packages/composer-playground/e2e/data/sample-networks/import-network/lib/sample.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Sample transaction processor function.
+ * @param {org.acme.sample.SampleTransaction} tx The sample transaction instance.
+ * @transaction
+ */
+function sampleTransaction(tx) {
```
Comment (sstone1) [inline packages/composer-playground/e2e/data/sample-networks/import-network/lib/sample.js]: y u no pull in ES7/ES8 version of sample with async/await goodness?! :trollface:

File: packages/composer-playground/e2e/utils/code-runner.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            let stderr;
+
+            return new Promise((resolve, reject) => {
+                let childCliProcess = exec(cmd);
+
+                childCliProcess.stdout.setEncoding('utf8');
+                childCliProcess.stderr.setEncoding('utf8');
+
+                childCliProcess.stdout.on('data', (data) => {
+                    stdout += data;
```
Comment (sstone1) [inline packages/composer-playground/e2e/utils/code-runner.ts]: y u no print output?

File: packages/composer-playground/e2e/utils/code-runner.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                });
+
+                childCliProcess.stderr.on('data', (data) => {
+                    stderr += data;
```
Comment (sstone1) [inline packages/composer-playground/e2e/utils/code-runner.ts]: y u no print output?

File: packages/composer-tests-integration/scripts/run-integration-tests.sh [hunk 5]
```diff
@@ -145,9 +142,9 @@ for INTEST in $(echo ${INTEST} | tr "," " "); do
 
     # Run the integration tests.
     if [[ ${INTEST} == *nohsm ]]; then
-        npm run int-test-nohsm 2>&1 | tee
+        npm run int-test-nohsm 2>&1 || :
```
Comment (sstone1) [inline packages/composer-tests-integration/scripts/run-integration-tests.sh]: wot is this `:`

File: packages/composer-playground/e2e/test.ts [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+import fs = require('fs');
+
+import { FabricHelper } from './utils/fabric-helper';
+import { CodeRunner } from './utils/code-runner';
+import { Constants } from './constants';
+import { ComposerUtils } from './utils/composer-utils';
+
+// Build test BNAs from sample networks and then proceed with test
+let networks = Constants.sampleNetworks;
+networks.reduce((chainPromise, network) => {
```
Comment (sstone1) [inline packages/composer-playground/e2e/test.ts]: could use async/await and `for ... of` for loop niceness

File: packages/composer-playground/e2e/test.ts [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+        } else {
+            return ComposerUtils.buildArchive(networkPath, bnaFile);
+        }
+    });
+}, Promise.resolve())
+.then(() => {
+    // Are we running against fabric?
+    const args = process.argv;
+    if (args.indexOf('--fabric') !== -1 || args.indexOf('-f') !== -1) {
+        runFabric();
```
Comment (sstone1) [inline packages/composer-playground/e2e/test.ts]: y u no `return`

File: packages/composer-playground/e2e/test.ts [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+});
+
+function runFabric() {
+    console.log('Setting up fabric, one moment...');
+    return FabricHelper.start()
+    .then (() => {
+        return FabricHelper.createPeerAdmin();
+    })
+    .then (() => {
+        runTests('fabric');
```
Comment (sstone1) [inline packages/composer-playground/e2e/test.ts]: y u no `return`

File: packages/composer-playground/e2e/test.ts [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+        }
+    });
+}, Promise.resolve())
+.then(() => {
+    // Are we running against fabric?
+    const args = process.argv;
+    if (args.indexOf('--fabric') !== -1 || args.indexOf('-f') !== -1) {
+        runFabric();
+    } else {
+        runTests('web');
```
Comment (sstone1) [inline packages/composer-playground/e2e/test.ts]: y u no `return`

File: packages/composer-tests-integration/scripts/run-integration-tests.sh (hunk 5 above)
Comment (nklincoln) [reply packages/composer-tests-integration/scripts/run-integration-tests.sh]: always pass

File: packages/composer-playground/e2e/utils/code-runner.ts [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { exec } from 'child_process';
+import { stripAnsi } from 'strip-ansi';
```
Comment (sstone1) [inline packages/composer-playground/e2e/utils/code-runner.ts]: y u no add to `package.json`?!

**Pull request merged**